torch/_functorch/fx_minifier.py

```diff
@@ -7,46 +7,115 @@
 from functools import wraps, partial
 from dataclasses import dataclass
 from .compile_utils import get_placeholders, get_outputs
+from torch.utils._content_store import ContentStoreWriter
+from torch.hub import tqdm
+from torch.multiprocessing.reductions import StorageWeakRef
+import os.path
+
+is_tuple = object()
+
+@dataclass
+class LoadTensorMeta:
+    size: List[int]
+    stride: List[int]
+    dtype: torch.dtype
+    device: torch.device
 
 class ConcreteProp(torch.fx.Interpreter):
-    def run_node(self, n):
-        result = super().run_node(n)
+    def __init__(self, mod, *, writer=None, skip_offload=False):
+        super().__init__(mod)
+        self.writer = writer
+        self.skip_offload = skip_offload
```

Comment:
Still reading, but what's the purpose of passing in a ContentStoreWriter, but setting `skip_offload=True`?

Reply:
Oh I think I see:

(1) `--offload_to_disk=True --skip-saving-eager-intermediates` won't actually offload anything, but every time the minifier asks for an intermediate, we'll generate random data on the fly.

(2) contrast that to the original behavior of the minifier, which is that it would save every intermediate tensor in (gpu) memory.

So the benefit is that we're as memory-efficient as `offload_to_disk` normally is, but we just get random data. Does that sound right? (that combo of flags is a bit confusing).

Reply:
Not quite.

(1) This won't write anything to disk, because we assume you *already* wrote everything to disk; we'll use the preexisting data on disk. This is very useful because on the really big models, the offloading is quite time consuming, and if you're iterating on minifier fixes you don't want to wait for the offload each time.

Reply:
ah thanks!